CMakeLists.txt

```
@@ -31,15 +31,24 @@ find_package(Threads REQUIRED)
 target_link_libraries(fdeep INTERFACE Threads::Threads)
 
 hunter_add_package(FunctionalPlus) # default noop
-find_package(FunctionalPlus CONFIG REQUIRED)
-target_link_libraries(fdeep INTERFACE FunctionalPlus::fplus)
-
-hunter_add_package(Eigen) # default noop
-find_package(Eigen3 CONFIG REQUIRED)
+if(TARGET fdeep)                   # if used via subdirectory
+  # Note: Sadely, FunctionalPlus does not provide an alias for use as a subdirectory
+  target_link_libraries(fdeep INTERFACE fplus)
+else()
+  find_package(FunctionalPlus CONFIG REQUIRED)
+  target_link_libraries(fdeep INTERFACE FunctionalPlus::fplus)
+(endif)
```

```suggestion
endif()
```

I think there's a _very_ small typo here